### ISSUE TITLE
Add support for configurable SPI mode

### ISSUE THREAD

**Issue by tuupola** [tuupola/hagl_pico_mipi#8]
See documentation for [spi_set_format()](https://raspberrypi.github.io/pico-sdk-doxygen/group__hardware__spi.html#ga137168d01d302239d52420ad4f895d44) for information.

**Comment by timskillman** [comment]
Yes, this would be great as I have several IPS screens that require SPI mode 3 support (clock polarity=1 / clock phase=1)
